File: algorithms/119_Pascal'sTriangleII.py
class Solution:
    def getRow(self, rowIndex: 'int') -> 'List[int]':
        """
        same implementation as the java version, but using python, the syntax is much easier to understand
        """
        lastList = [1]
        if rowIndex == 0: return lastList
        
        for i in range(1, rowIndex+1):
            lastList = [1] + [lastList[j-1] + lastList[j] for j in range(1, i)] + [1]
        return lastList
        

File: algorithms/Math/119_Pascal'sTriangleII.java
class Solution {
    public List<Integer> getRow(int rowIndex) {
        List<Integer> lastList = new LinkedList<Integer>(), list;
        lastList.add(1);
        if (rowIndex == 0) return lastList;
        
        for (int i = 1; i < rowIndex + 1; ++i) {
            list = new LinkedList<Integer>();
            list.add(1);
            for (int j = 1; j < i; ++j) {
                list.add(lastList.get(j-1) + lastList.get(j));
            }
            list.add(1);
            lastList = list;
        }
        return lastList;
    }
}